src/components/interface/cFE/cFE_emu.h

```diff
@@ -0,0 +1,106 @@
+#include <cos_types.h>
+
+#include <cfe_evs.h>
+
+#define EMU_BUF_SIZE 512
+
+/* TODO: Alphabetize me! */
+union shared_region {
+	struct {
+		CFE_EVS_BinFilter_t filters[CFE_EVS_MAX_EVENT_FILTERS];
+		uint16              NumEventFilters;
+		uint16              FilterScheme;
+	} cfe_evs_register;
+	struct {
+		CFE_SB_PipeId_t PipeId;
+		uint16          Depth;
+		char            PipeName[OS_MAX_API_NAME];
+	} cfe_sb_createPipe;
+	struct {
+		char           MsgBuffer[EMU_BUF_SIZE];
+		CFE_SB_MsgId_t MsgId;
+		uint16         Length;
+		boolean        Clear;
+	} cfe_sb_initMsg;
+	struct {
+		char   Msg[EMU_BUF_SIZE];
+		uint16 EventID;
+		uint16 EventType;
+	} cfe_evs_sendEvent;
+	struct {
+		uint32 RunStatus;
+	} cfe_es_runLoop;
+	struct {
+		CFE_SB_PipeId_t PipeId;
+		int32           TimeOut;
+		char            Msg[EMU_BUF_SIZE];
+	} cfe_sb_rcvMsg;
+	struct {
+		CFE_SB_Msg_t Msg;
+	} cfe_sb_getMsgLen;
+	struct {
+		char Msg[EMU_BUF_SIZE];
+	} cfe_sb_msg;
+	CFE_TIME_SysTime_t time;
+	struct {
+		char PrintBuffer[CFE_TIME_PRINTED_STRING_SIZE];
+		CFE_TIME_SysTime_t TimeToPrint;
+	} cfe_time_print;
+	struct {
+		char Msg[EMU_BUF_SIZE];
+		uint16 CmdCode;
+	} cfe_sb_setCmdCode;
+	struct {
+		CFE_TIME_SysTime_t Time1;
+		CFE_TIME_SysTime_t Time2;
+		CFE_TIME_SysTime_t Result;
+	} cfe_time_add;
+	struct {
+		CFE_TIME_SysTime_t Time1;
+		CFE_TIME_SysTime_t Time2;
+		CFE_TIME_Compare_t Result;
+	} cfe_time_compare;
+	struct {
+		CFE_ES_TaskInfo_t TaskInfo;
+		uint32 TaskId;
+	} cfe_es_getTaskInfo;
+	struct {
+		uint32 ResetSubtype;
+	} cfe_es_getResetType;
+	struct {
+		uint32 CounterId;
+		uint32 Count;
+	} cfe_es_getGenCount;
+	struct {
+		uint32 CounterId;
+		char CounterName[EMU_BUF_SIZE];
+	} cfe_es_getGenCounterIDByName;
+};
+
+int emu_backend_request_memory(spdid_t client);
+
+int32 emu_CFE_ES_GetGenCount(spdid_t client);
```

Comment:
You should use the `cos_inv_token()` in the server component for the client's spdid instead.